src/setup/setup.test.ts

```diff
@@ -1,9 +1,75 @@
-import * as Setup from './setup';
-describe('setup functions', () => {
-	it('should set the global variable when passed', () => {
-		const APIKEY = 'HELLO';
-		Setup.BreinifySetup({ apiKey: APIKEY });
+describe('Setup functions', () => {
+	beforeEach(() => {
+		jest.resetModules();
+	});
+
+	it('valid apiKey, valid secret', () => {
+		const Setup = require('./setup');
+		const API_KEY = 'API_KEY';
+		const SECRET = 'SECRET';
+		Setup.BreinifySetup({ apiKey: API_KEY, secret: SECRET });
+
+		expect(Setup.BreinifyGlobalConfigs).toEqual({ apiKey: API_KEY, secret: SECRET });
+	});
+
+	it('valid apiKey, no secret', () => {
+		const Setup = require('./setup');
+		const API_KEY = 'API_KEY';
+		Setup.BreinifySetup({ apiKey: API_KEY });
+
+		expect(Setup.BreinifyGlobalConfigs).toEqual({ apiKey: API_KEY, secret: undefined });
+	});
+
+	it('no apiKey, valid secret', () => {
```

Comment:
NIT: maybe we should describe the behavior of the test here. Maybe something like 
`it('should throw when no api key is passed in', ...)`

What do you think?

Reply:
I like that idea